File: linear-search/linear-search.cs
using System;

class LinearSearch {
    static int Search(int[] arr, int target) {
        for (int i = 0; i < arr.Length; i++) {
            if (arr[i] == target) return i;
        }
        return -1;
    }
    
    static void Main() {
        int[] arr = {1, 2, 3, 4, 5};
        Console.WriteLine(Search(arr, 3));
    }
}

File: linear-search/linear-search.cpp
#include <iostream>
#include <vector>

int linearSearch(const std::vector<int>& arr, int target) {
    for (int i = 0; i < arr.size(); i++) {
        if (arr[i] == target) return i;
    }
    return -1;
}

int main() {
    std::vector<int> arr = {5, 2, 8, 1, 9, 3};
    std::cout << linearSearch(arr, 8) << std::endl;
    std::cout << linearSearch(arr, 7) << std::endl;
    return 0;
}
